app/controllers/books_controller.rb

```diff
@@ -0,0 +1,86 @@
+class BooksController < ApplicationController
+  def index
+    @books = Book.all.order('rank desc, id')
+  end
+
+  def find_book
+    if Book.exists?(params[:id].to_i) == true
+      return Book.find(params[:id].to_i)
+    else
+      render :status => 404
```

Comment:
Because of the way we're using this as a helper method we should (a) move it to the bottom of the controller under a `private` section and (b) make it a controller filter by using `before_action`.

Right now we're actually returning the value of `render :status => 404` into our controller actions such as `show` when the requested book does not exist, which then gets stored in the `@mybook` variable.

Later, when the action attempts to `redirect_to` or `render` on its own (or by the default Rails behavior) it will cause an error because we've already called `render`.

If we instead use this as a controller filter Rails will detect that we've called `render` and not run the action (`show`, `upvote`, etc.) at all.